pkg/controller/operators/catalog/operator.go

```diff
@@ -1389,25 +1389,23 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 	ref, err := querier()
 	if err != nil {
 
-		// Retry sync if non-fatal error
-		if !scoped.IsOperatorGroupError(err) {
-			syncError = fmt.Errorf("attenuated service account query failed - %v", err)
-			return
-		}
-
-		// Mark the InstallPlan as failed for a fatal Operator Group related error
-		logger.Infof("attenuated service account query failed - %v", err)
-		ipFailError := fmt.Errorf("invalid operator group - %v", err)
+		// Set status condition/message and retry sync if any error
+		ipFailError := fmt.Errorf("attenuated service account query failed - %v", err)
+		logger.Infof(ipFailError.Error())
 
-		if err := o.transitionInstallPlanToFailed(plan, logger, v1alpha1.InstallPlanReasonInstallCheckFailed, ipFailError.Error()); err != nil {
-			// retry for failure to update status
+		now := o.now()
+		out := plan.DeepCopy()
+		out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled,
```

Comment:
Is this condition reset anywhere after the OperatorGroup issue is resolved, but before all plan steps succeed?

Reply:
The next step is bundle unpacking, and if any error is encountered there, this condition is reset there (tested [here](https://github.com/operator-framework/operator-lifecycle-manager/pull/2215/files#R3245)). The step right after that is to indicate that all plan steps have succeeded. So the possible unhappy paths looks covered.

Reply:
Bundle unpacking could take a long time (several minutes, timing out after 10 minutes by default), and plan step errors are retried for up to 1 minute (by default). It could be confusing for users who see the OperatorGroup error, resolve the problem, then continue to see the same error for another 5 minutes.

Reply:
@benluddy updated the PR to reset the condition, along with a note about assumptions made.